components/Common/ShellBox/index.tsx

```diff
@@ -0,0 +1,41 @@
+import { useRef } from 'react';
+import { FormattedMessage } from 'react-intl';
+import styles from './index.module.scss';
+import { useCopyToClipboard } from '../../../hooks/useCopyToClipboard';
+import type { FC, PropsWithChildren, MouseEvent, ReactNode } from 'react';
+
+type ShellBoxProps = {
+  children: string | ReactNode;
+  textToCopy?: string;
+};
+
+const ShellBox: FC<PropsWithChildren<ShellBoxProps>> = ({
+  children,
+  textToCopy,
+}: PropsWithChildren<ShellBoxProps>) => {
+  const [copied, copyText] = useCopyToClipboard();
+
+  const shellBoxRef = useRef<HTMLElement>(null);
+
+  const handleCopyCode = async (event: MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    await copyText(textToCopy || shellBoxRef.current?.innerHTML);
```

Comment:
If we want to make a "hack" to remove `$`

```suggestion
    // By default we want to use the textToCopy props but if it's absent
    // we allow the user to copy by getting the inner HTML content of the Element
		const _textToCopy = textToCopy || shellBoxRef.current?.innerHTML || '';

    await copyText(_textToCopy.replace('$', '');
```

Reply:
I know @AugustinMauroy you said you had no idea how to accomplish this, but here's a simple solution.